pkg/ddc/alluxio/ufs_internal.go

```diff
@@ -96,6 +96,148 @@ func (e *AlluxioEngine) shouldMountUFS() (should bool, err error) {
+			continue
+		}
+		alluxioPathHaveMountted := utils.UFSPathBuilder{}.GenAlluxioMountPath(mount, dataset.Status.Mounts)
+		resultHaveMounted = append(resultHaveMounted, alluxioPathHaveMountted)
+	}
+
+	return resultInCtx, resultHaveMounted, err
+
+}
+
+func (e *AlluxioEngine) calculateMountPointsChanges(mountsHaveMountted []string, mountsInContext []string) ([]string, []string) {
+	removed := []string{}
+	added := []string{}
+
+	for _, v := range mountsHaveMountted {
+		if ContainsString(mountsInContext, v) == false {
+			removed = append(removed, v)
+		}
+	}
+
+	for _, v := range mountsInContext {
+		if ContainsString(mountsHaveMountted, v) == false {
+			added = append(added, v)
+		}
+	}
+
+	return added, removed
+}
+
+// ContainsString returns true if a string is present in a iteratee.
+func ContainsString(s []string, v string) bool {
+	for _, vv := range s {
+		if vv == v {
+			return true
+		}
+	}
+	return false
+}
+
+func (e *AlluxioEngine) processUpdatingUFS(updatedUFSMap map[string][]string) (err error) {
+	dataset, err := utils.GetDataset(e.Client, e.name, e.namespace)
+	if err != nil {
+		return err
+	}
+
+	podName, containerName := e.getMasterPodInfo()
+	fileUitls := operations.NewAlluxioFileUtils(podName, containerName, e.namespace, e.Log)
+
+	ready := fileUitls.Ready()
+	if !ready {
+		return fmt.Errorf("the UFS is not ready")
+	}
+
+	// Iterate all the mount points, do mount if the mount point is in added array
+	for _, mount := range dataset.Spec.Mounts {
+		if common.IsFluidNativeScheme(mount.MountPoint) {
+			continue
+		}
+
+		alluxioPath := utils.UFSPathBuilder{}.GenAlluxioMountPath(mount, dataset.Spec.Mounts)
+		added := updatedUFSMap["added"]
+		if len(added) > 0 && utils.ContainsString(added, alluxioPath) {
+			mountOptions := map[string]string{}
+			for key, value := range mount.Options {
+				mountOptions[key] = value
+			}
+
+			// Configure mountOptions using encryptOptions
+			// If encryptOptions have the same key with options, it will overwrite the corresponding value
+			for _, encryptOption := range mount.EncryptOptions {
+				key := encryptOption.Name
+				secretKeyRef := encryptOption.ValueFrom.SecretKeyRef
+
+				secret, err := utils.GetSecret(e.Client, secretKeyRef.Name, e.namespace)
+				if err != nil {
+					e.Log.Info("can't get the secret")
+					return err
+				}
+
+				value := secret.Data[secretKeyRef.Key]
+				e.Log.Info("get value from secret")
+
+				mountOptions[key] = string(value)
+			}
+			err = fileUitls.Mount(alluxioPath, mount.MountPoint, mountOptions, mount.ReadOnly, mount.Shared)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	// unmount the mount point in the removed array
+	removed := updatedUFSMap["removed"]
+	if len(removed) > 0 {
+		for _, mount_remove := range removed {
+			err = fileUitls.UnMount(mount_remove)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	err = e.SyncMetadata()
```

Comment:
I think there is a missing step here. The `SyncMetadata()` func won't actually take effect because it firstly checks the Dataset's `UfsTotal` status(check [here](https://github.com/fluid-cloudnative/fluid/blob/master/pkg/ddc/alluxio/metadata.go#L80)). To sync metadata, we should first modify the `UfsTotal` status to empty.

Reply:
Maybe modifing the `UfsTotal` status is still not enough. `SyncMetadata` also do es metadata restoration. When we want to resync metadata on mount changes, there is no need to do metadata restoration.